packages/components/src/Combobox/Combobox.tsx

```diff
@@ -1,191 +1,71 @@
-  const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
-
-  const filteredOptions = props.options.filter(option =>
-    option.toLowerCase().includes(searchValue.toLowerCase()),
-  );
-
-  function handleSearch(event: React.ChangeEvent<HTMLInputElement>) {
-    setSearchValue(event.target.value);
-  }
+function useComboboxValidation(children: ReactNode): {
+  renderTrigger: ReactNode;
+  renderContent: ReactNode;
+} {
+  const childrenArray = React.Children.toArray(children);
+  let renderTrigger: ReactNode, renderContent: ReactNode;
+
+  childrenArray.forEach(child => {
+    if (isTriggerElement(child)) {
+      if (renderTrigger) {
+        useAssert(true, "Combobox can only have one Trigger element");
+      }
+      renderTrigger = child;
+    }
+    if (isContentElement(child)) {
+      renderContent = child;
+    }
+  });
 
-  function handleSelection(event: React.MouseEvent<HTMLButtonElement>) {
-    const selected = event.currentTarget.innerText;
-    setSelectedOption(selected);
-    setOpen(false);
-    setSearchValue("");
-    props.onSelection(selected);
+  if (!renderTrigger || !renderContent) {
+    useAssert(true, "Combobox must have a Trigger and Content element");
   }
-  return (
-    <div className={classnames(styles.content, !open && styles.hidden)}>
-      <div className={styles.search}>
-        <input
-          type="search"
-          className={styles.searchInput}
-          placeholder="Search teammates"
-          onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
-            handleSearch(event)
-          }
-          value={searchValue}
-        />
-
-        {/* if there is a search value show the clear icon */}
-        {searchValue && (
-          <div
-            className={styles.triggerIcon}
-            onClick={() => setSearchValue("")}
-            role="button"
-          >
-            <Icon name="remove" size="small" />
-          </div>
-        )}
-      </div>
-
-      {filteredOptions.map(option => (
-        <button
-          className={classnames(
-            option === selectedOption && styles.selectedOption,
-          )}
-          role="button"
-          tabIndex={0}
-          key={option}
-          onClick={handleSelection}
-        >
-          {option}
-        </button>
-      ))}
 
-      {/* if there are no results matching the search value show no results item */}
-      {filteredOptions.length === 0 && <p>No results for {searchValue}</p>}
+  return {
+    renderContent,
+    renderTrigger,
+  };
+}
 
-      {/* only render actions if their are any */}
-      {props.children && <div className={styles.actions}>{props.children}</div>}
-    </div>
+function isTriggerElement(child: ReactNode): boolean {
+  return (
+    React.isValidElement(child) &&
+    typeof child !== "string" &&
+    typeof child.type !== "string" &&
+    (child.type.name == "TriggerButton" || child.type.name == "TriggerChip")
```

Comment:
can also skip the string check and match it with the component itself

```suggestion
    React.isValidElement(child) &&
    (child.type === TriggerButton || child.type === TriggerChip)
```

Reply:
what in the world. I swear I tried that at one point and it did not work, but yeah you're right that does the trick!